docs: add section on installing rustup and wasm32 target (#49)
extism/rust-pdk: README.md

crate_type = ["cdylib"]
```

### Rustup and wasm32-unknown-unknown installation

> Our example below will use of the wasm32-unknown-unknown target. If this is not installed you will need to do so before this example will build. The easiest way to do this is use [`rustup`](https://rustup.rs/).

Reviewer: ```suggestion
Our example below will use the `wasm32-unknown-unknown` target. If this is not installed you will need to do so before this example will build. The easiest way to do this is use [`rustup`](https://rustup.rs/).
```

> Our example below will use of the wasm32-unknown-unknown target. If this is not installed you will need to do so before this example will build. The easiest way to do this is use *rustup*. If you have *rustup* installed skip the next step.

Reviewer: ```suggestion
Our example below will use of the wasm32-unknown-unknown target. If this is not installed you will need to do so before this example will build. The easiest way to do this is use [`rustup`](https://rustup.rs/).
```

> ```toml

Reviewer: ```suggestion
```bash
```

curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh

> ```

Reviewer 2: The placement if this is a little strange because just a moment earlier, the user was running cargo, etc. So if they don't have rust installed they should be failing there right? Should we maybe move this up?

Reviewer: yea I'd support that

Author: Good catch. I thought about that but assumed they probably had rust/cargo installed but maybe not rustup. Total noob so wasn't sure if rustup was part of the typical rust install.

Author: Actually.. should the comment and ### Rustup.. part be up at the top too? Thus this whole bit is at the top? Otherwise just the curl command may look a little odd.

> Once *rustup* is installed, add the wasm32-unknown-unknown target:

Reviewer: ```suggestion
Once `rustup` is installed, add the `wasm32-unknown-unknown` target:
```

> ```toml

Reviewer: ```suggestion
```bash
```

rustup target add wasm32-unknown-unknown
```

> 

Reviewer: Sorry for the delay here, didn't realize this got left behind... 

> Actually.. should the comment and ### Rustup.. part be up at the top too? Thus this whole bit is at the top? Otherwise just the curl command may look a little odd.

Do you mean to include this code attached here to the `curl` command to be moved up?
